blockchain/src/chain_store.rs

```diff
@@ -2,93 +2,92 @@ use nimiq_account::RevertInfo;
 use nimiq_block::Block;
 use nimiq_blockchain_interface::{BlockchainError, ChainInfo, Direction};
 use nimiq_database::{
-    traits::{Database, ReadCursor, ReadTransaction, WriteCursor, WriteTransaction},
-    DatabaseProxy, TableFlags, TableProxy, TransactionProxy, WriteTransactionProxy,
+    declare_table,
+    mdbx::{MdbxDatabase, MdbxReadTransaction, MdbxWriteTransaction, OptionalTransaction},
+    traits::{Database, DupReadCursor, ReadCursor, ReadTransaction, WriteCursor, WriteTransaction},
 };
 use nimiq_hash::Blake2bHash;
 use nimiq_primitives::{policy::Policy, trie::trie_diff::TrieDiff};
 
+declare_table!(HeadTable, "Head", () => Blake2bHash);
```

Comment:
I think this is actually a breaking change. Sorry, I even suggested it to you. Perhaps we can break it out of this PR, e.g. create a new unit struct `Head` that serializes to and from `b"head"`?

Reply:
It would be in a new table anyway. Although, if we care about backwards compatibility, we can give both tables the same name, which allows accessing it with the two different types.